react/src/components/Chat/CustomModelForm.tsx

```diff
@@ -50,65 +57,48 @@ const CustomModelForm: React.FC<CustomModelFormProps> = ({
         style={{ flex: 1 }}
         key={baseURL}
         initialValues={{
-          baseURL: baseURL,
+          basePath: basePath,
           token: token,
         }}
       >
-        {alert ? (
-          <div style={{ marginBottom: themeToken.size }}>{alert}</div>
-        ) : null}
-        <Form.Item
-          label="baseURL"
-          name="baseURL"
-          rules={[
-            {
-              type: 'url',
-            },
-            {
-              required: true,
-            },
-          ]}
-        >
-          <Input placeholder="https://domain/v1" />
+        <div style={{ marginBottom: themeToken.size }}>
+          <Alert
+            type="warning"
+            showIcon
+            message={t('chatui.CannotFindModel')}
+          />
+        </div>
+        <Form.Item label={t('modelService.BasePath')} name="basePath">
+          <Input
+            placeholder="v1"
+            addonBefore={origin}
+            defaultValue={basePath}
+          />
         </Form.Item>
-        <Form.Item
-          label="Model ID"
-          name="modelId"
-          rules={[
-            {
-              required: true,
-            },
-          ]}
-        >
-          <Input placeholder="llm-model" />
-        </Form.Item>
-        <Form.Item label="Token" name="token">
-          <Input />
+        <Form.Item label={t('modelService.Token')} name="token">
+          <EndpointTokenSelect
+            loading={isPendingSubmit}
+            endpointId={endpointId}
+          />
         </Form.Item>
+        <Button
+          icon={<ReloadOutlined />}
+          loading={isPendingSubmit}
+          onClick={() => {
+            startSubmitTransition(() => {
+              const basePath = formRef.current?.getFieldValue('basePath') ?? '';
+              onSubmit?.({
+                baseURL: new URL(basePath, origin).toString(),
+                token: formRef.current?.getFieldValue('token'),
+              });
+            });
+          }}
+        >
```

Comment:
Since the `onSubmit` prop is determined by the parent using the `CustomModelForm`, it is better to decide whether to apply the transition within the `onSubmit` function where `CustomModelForm` is used, rather than within `CustomModelForm` itself. 
Additionally, let's add a `loading` prop to `CustomModelForm` so that the loading state can be determined by the parent.